weed/shell/command_volume_check_disk.go

```diff
@@ -43,6 +49,66 @@ func (c *commandVolumeCheckDisk) Help() string {
 `
 }
 
+func (c *commandVolumeCheckDisk) getVolumeStatusFileCount(vid uint32, dn *master_pb.DataNodeInfo) (fileCount, deletedFileCount uint64) {
+	_ = operation.WithVolumeServerClient(false, pb.NewServerAddressWithGrpcPort(dn.Id, int(dn.GrpcPort)), c.env.option.GrpcDialOption, func(volumeServerClient volume_server_pb.VolumeServerClient) error {
+		resp, _ := volumeServerClient.VolumeStatus(context.Background(), &volume_server_pb.VolumeStatusRequest{
+			VolumeId: uint32(vid),
+		})
+		if resp != nil {
+			fileCount = resp.FileCount
+			deletedFileCount = resp.FileDeletedCount
+		}
+		return nil
+	})
+	return fileCount, deletedFileCount
+}
+
+func (c *commandVolumeCheckDisk) eqVolumeFileCount(a, b *VolumeReplica) (bool, bool) {
+	var waitGroup sync.WaitGroup
+	var fileCountA, fileCountB, fileDeletedCountA, fileDeletedCountB uint64
+	waitGroup.Add(1)
+	go func() {
+		defer waitGroup.Done()
+		fileCountA, fileDeletedCountA = c.getVolumeStatusFileCount(a.info.Id, a.location.dataNode)
+	}()
+	waitGroup.Add(1)
+	go func() {
+		defer waitGroup.Done()
+		fileCountB, fileDeletedCountB = c.getVolumeStatusFileCount(b.info.Id, b.location.dataNode)
+	}()
+	// Trying to synchronize a remote call to two nodes
+	waitGroup.Wait()
+	return fileCountA == fileCountB, fileDeletedCountA == fileDeletedCountB
+}
+
+func (c *commandVolumeCheckDisk) doSkipVolume(a, b *VolumeReplica, timeBeforePulse int64, syncDeletions, verbose bool, writer io.Writer) bool {
+	doSyncTotalFileCount := false
+	doSyncDeletedCount := false
+	if syncDeletions && a.info.DeleteCount != b.info.DeleteCount {
+		doSyncDeletedCount = true
+	}
+	if a.info.FileCount != b.info.FileCount {
+		doSyncTotalFileCount = true
+	}
+	if doSyncTotalFileCount || doSyncDeletedCount {
+		if timeBeforePulse > a.info.ModifiedAtSecond && timeBeforePulse > b.info.ModifiedAtSecond {
+			return false
```

Comment:
> If the volume modification time is close to the time of the last statistics collection pulse in the topology, then additionally check the number of vils at the moment for each such volume

Here the condition is the modification times are not close to stats collection time.

Reply:
Yes, there is inverted logic.

Added a comment to the code for understanding.